src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockMigrationService.cs

```diff
@@ -185,26 +204,47 @@ private async Task SaveChanges(DataBlock dataBlock, bool dryRun)
             }
         }
 
-        private async Task<DateTime> GetDataBlockCreatedDate(DataBlock dataBlock)
+        private async Task<DateTime?> GetDataBlockCreatedDate(DataBlock dataBlock)
         {
-            DateTime createdDate;
-
-            if (dataBlock.Created != null)
+            try
             {
-                createdDate = dataBlock.Created.Value;
+                if (dataBlock.Created != null)
+                {
+                    return dataBlock.Created.Value;
+                }
+
+                var owningReleaseByReleaseContentBlocks = (await _contentDbContext
+                        .ReleaseContentBlocks
+                        .Include(rcb => rcb.Release)
+                        .SingleOrDefaultAsync(rcb => rcb.ContentBlockId == dataBlock.Id))
+                    ?.Release;
+
+                if (owningReleaseByReleaseContentBlocks != null)
+                {
+                    return owningReleaseByReleaseContentBlocks.Created;
+                }
+
+                var owningReleaseByContentSection = (await _contentDbContext
+                        .ReleaseContentSections
+                        .Include(rcs => rcs.Release)
+                        .Include(rcs => rcs.ContentSection)
+                        .ThenInclude(cs => cs.Content)
+                        .Where(rcs => rcs.ContentSection.Content.Contains(dataBlock))
+                        .FirstOrDefaultAsync())
+                    ?.Release;
+
+                if (owningReleaseByContentSection != null)
+                {
+                    return owningReleaseByContentSection.Created;
+                }
```

Comment:
I think the name `ReleaseContentBlock` has been a bit misleading as we don't link any other types of `ContentBlock` through it apart from data blocks. So to find the Release for a datablock _if_ it's been linked (plenty of data blocks aren't - many are created just to provide table highlights or remain unused), we'd lookup the Release through `ReleaseContentBlock`. Is it necessary to look for them in `ReleaseContentSection` as a fallback? I guess no harm in doing so but maybe we should log this to find out why a data block is in a ContentSection without a link in ReleaseContentBlock. If it ever happened nobody would be able to load the table fast track link.